src/ipc-wrapper/server.js

```diff
@@ -0,0 +1,78 @@
+// @ts-check
+import { createServer } from 'rpc-reflector'
+import { MANAGER_CHANNEL_ID, MAPEO_IPC_ID, SubChannel } from './sub-channel.js'
+
+/**
+ * @param {import('../mapeo-manager.js').MapeoManager} manager
+ * @param {import('./sub-channel.js').MessagePortLike} messagePort
+ */
+export function createMapeoServer(manager, messagePort) {
+  /** @type {Map<string, { close: () => void, channel: SubChannel }>} */
+  const existingProjectServers = new Map()
+
+  const mapeoIpcChannel = new SubChannel(messagePort, MAPEO_IPC_ID)
+  mapeoIpcChannel.start()
+
+  mapeoIpcChannel.on('message', handleGetProject)
```

Comment:
i changed the implementation such that it's only doing work when the client specifically requests it i.e. we call `client.getProject()`, which sends a subchannel message from the client that is received here. 

before, we were listening on the whole messagePort and then filtering for relevant messages. this approach handles "indirect" requests of a project client i.e. a client calls a method on an active project client, but I'm not sure that's a situation we actually run into (since we always keep a map of existing project clients and don't do any automatic cleanup).

wondering if the edge case here is still relevant or if this new approach suffices

Reply:
reverted 👍